packages/smooth_app/lib/pages/hunger_games/question_page.dart

```diff
@@ -205,70 +96,94 @@ class _QuestionPageState extends State<_QuestionPage>
   }
 
   void _loadNextQuestions() {
-    _loadQuestions(
-      request: _questionsQuery.getQuestions(
-        _localDatabase,
-        _numberQuestionsNext,
-      ),
-    );
-    _currentQuestionIndex = 0;
+    try {
+      setState(() {
+        _state = const _RobotoffQuestionLoadingState();
+      });
+      _loadQuestions(
+        request: _questionsQuery.getQuestions(
+          _localDatabase,
+          _numberQuestionsNext,
+        ),
+      );
+      _currentQuestionIndex = 0;
+    } catch (e) {
+      debugPrint('Error loading next questions: $e');
+      _updateState(_RobotoffQuestionErrorState(Exception(e.toString())));
+    }
+  }
+
+  void _incrementQuestionsAnswered() {
```

Comment:
Cannot see the added value of moving the code here.
It was more relevant to put that in `_saveAnswer`, wasn't it?